pitop/core/mixins/digital_component_checks.py

```diff
@@ -0,0 +1,15 @@
+
+valid_digital_ports = ["D0", "D1", "D2", "D3", "D4", "D5", "D6", "D7"]
+
+
+class DigitalComponentChecks:
+    """Performs basic checks on validity of user-specified port."""
+
+    def __init__(self, port_name):
+        self.port_name = port_name
+
+        if port_name not in valid_digital_ports:
```

Comment:
I think I would prefer to see regex used here to match ports: `D[0-7]`

Reply:
good idea